dataland-backend/src/main/kotlin/org/dataland/datalandbackend/repositories/CompanyIdentifierRepository.kt

```diff
@@ -2,9 +2,27 @@ package org.dataland.datalandbackend.repositories
 
 import org.dataland.datalandbackend.entities.CompanyIdentifierEntity
 import org.dataland.datalandbackend.entities.CompanyIdentifierEntityId
+import org.dataland.datalandbackend.entities.StoredCompanyEntity
+import org.dataland.datalandbackend.model.enums.company.IdentifierType
 import org.springframework.data.jpa.repository.JpaRepository
 
 /**
  * A JPA interface for accessing the CompanyIdentifier Entity
  */
-interface CompanyIdentifierRepository : JpaRepository<CompanyIdentifierEntity, CompanyIdentifierEntityId>
+interface CompanyIdentifierRepository : JpaRepository<CompanyIdentifierEntity, CompanyIdentifierEntityId> {
```

Comment:
What is the reason that these two functions have appeared now as part of the PUT/PATCH story?

Reply:
The `deleteAllByCompanyAndIdentifierType()` is used to delete all identifiers of the company for a specific identifier type (e.g. all Leis) from the database before writing the "updated" Leis to the database.
It is used for the `PATCH` endpoint. You can see it kind of like a PUT on a identifier-type-level.

The `deleteAllByCompany()` is used to delete all identifiers of the company from the database before writing the "updated" identifiers to the database.
It is used for the `PUT` endpoint to erase all existing identifiers, before writing the new ones.